tasks/lint.yaml

```diff
@@ -159,56 +159,92 @@ tasks:
   - name: license
     description: Lint for the SPDX license identifier being in source files
     actions:
-      - description: Checking for SPDX headers
+      - description: Checking for SPDX headers using YAML config with defaults
         shell:
           darwin: bash
           linux: bash
         cmd: |
-          # check for addlicense bin
-          if [ -x "$HOME/go/bin/addlicense" ]; then
-              echo "addlicense installed in $HOME/go/bin"
+          CONFIG_FILE=".license_config.yaml"
+
+          # If the config file exists, parse its values; otherwise, use default values.
+          if [ -f "$CONFIG_FILE" ]; then
+            LICENSE=$(uds zarf tools yq e '.license' "$CONFIG_FILE")
+            COPYRIGHT=$(uds zarf tools yq e '.copyright' "$CONFIG_FILE")
           else
-              echo "Error: addlicense is not installed in $HOME/go/bin" >&2
-              exit 1
+            LICENSE="AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial"
+            COPYRIGHT="Defense Unicorns"
+          fi
+
+          # Exit if license or copyright are not provided.
+          if [ -z "$LICENSE" ] || [ "$LICENSE" = "null" ] || [ -z "$COPYRIGHT" ] || [ "$COPYRIGHT" = "null" ]; then
+            echo "License or copyright configuration not provided; exiting."
+            exit 0
+          fi
+
+          # Check for addlicense binary
+          if [ ! -x "$HOME/go/bin/addlicense" ]; then
+            echo "Error: addlicense is not installed in $HOME/go/bin" >&2
+            exit 1
           fi
 
+          # Build ignore flags if the config file exists
           ignore_flags=()
-          while IFS= read -r line; do
-            # Skip comments and empty lines
-            [ -z "$line" ] || [[ "$line" =~ ^# ]] && continue
-            # Add the line to ignore_flags array with quotes around it
-            ignore_flags+=("-ignore" "\"$line\"")
-          done < .gitignore
+          if [ -f "$CONFIG_FILE" ]; then
+            while IFS= read -r pattern; do
+              [ -z "$pattern" ] && continue
+              ignore_flags+=("-ignore" "\"$pattern\"")
+            done < <(uds zarf tools yq e '.ignore[]' "$CONFIG_FILE")
+          fi
```

Comment:
I'm okay adding the new `ignore` option here in the config file but:
1. This ignore option would only be available if the end user is using a config file, which I think is _OK_ just means they have to explicitly add the 🦄 dual-license instead of using the default.
2. This removes the behavior of ignoring files from `.gitignore` automatically - forcing users to specify those same files in the new license config file. Unfortunately addlicense doesn't automatically ignore those which can be a pain for UDS Packages.

Could we keep the previous `.gitignore` behavior for backwards-compatibility? Otherwise I see this as a "breaking change" since users may be forced to create that config file and add all their gitignores to it.